transactions/bitcoin/context.ts

```diff
@@ -743,16 +624,22 @@ export class TransactionContext {
 
   async getUtxoFallbackToExternal(
     outpoint: string,
-  ): Promise<{ extendedUtxo?: ExtendedUtxo; addressContext?: AddressContext }> {
+  ): Promise<{ extendedUtxo?: ExtendedUtxo; addressContext?: AddressContext } | undefined> {
     const utxoData = await this.getUtxo(outpoint);
-
     if (utxoData.extendedUtxo) {
       return utxoData;
     }
 
-    const extendedUtxo = await this.paymentAddress.getExternalUtxo(outpoint);
-
-    return { extendedUtxo };
+    try {
+      const extendedUtxo = await this.paymentAddress.getExternalUtxo(outpoint);
+      return { extendedUtxo };
+    } catch (err) {
+      if (err.response && err.response.status === 404) {
+        return {};
+      } else {
+        return undefined;
+      }
+    }
```

Comment:
This implementation has the potential to silently fail (e.g. response is actual valid failure like a 429 or 504. We would end up showing the user incorrect info). We need to rethrow the error instead.

```suggestion
    } catch (err) {
      if (err.response && err.response.status === 404) {
        return {};
      } 
      
      throw err;
    }
```